src/main/java/org/checkerframework/specimin/UnsolvedSymbolVisitor.java

```diff
@@ -706,14 +703,12 @@ public Visitable visit(ObjectCreationExpr newExpr, Void p) {
     try {
       List<String> argumentsCreation = getArgumentsFromObjectCreation(newExpr);
       UnsolvedMethod creationMethod = new UnsolvedMethod("", type, argumentsCreation);
-      if (classAndPackageMap.containsKey(type)) {
-        UnsolvedClass newClass = new UnsolvedClass(type, classAndPackageMap.get(type));
-        newClass.addMethod(creationMethod);
-        this.updateMissingClass(newClass);
-      } else {
-        throw new RuntimeException("Unexpected class: " + type);
-      }
-
+      // we assume that an unsolved class not found among import statements should be in the same
+      // package as the current class
+      String packageName = classAndPackageMap.getOrDefault(type, currentPackage);
+      UnsolvedClass newClass = new UnsolvedClass(type, packageName);
+      newClass.addMethod(creationMethod);
```

Comment:
The code here, at lines 678-682 above, and at 529-533 above are all almost identical. We should move the common parts out into a method (whose signature should be something like `createUnsolvedClass(String className, String currentPackage)` and whose content would be lines 708, 709, and 711 here). Then, you can call this method at each of these locations. That will make future changes like this easier, by reducing coupling + duplication.